demo_nodes_cpp/src/topics/talker.cpp

```diff
@@ -54,16 +54,16 @@ class Talker : public rclcpp::Node
     // rclcpp::KeepAll{} if the user wishes.
     // (rclcpp::KeepLast(7) -> rclcpp::KeepAll() fails to compile)
     rclcpp::QoS qos(rclcpp::KeepLast{7});
-    pub_ = this->create_publisher<std_msgs::msg::String>("chatter", qos);
+    pub_ = this->create_publisher<example_interfaces::msg::String>("chatter", qos);
```

Comment:
this should resolve https://github.com/ros2/examples/pull/377#discussion_r1585622542